src/Internal/ProductAttributesLookup/LookupDataStore.php

```diff
@@ -0,0 +1,301 @@
+<?php
+/**
+ * LookupDataStore class file.
+ */
+
+namespace Automattic\WooCommerce\Internal\ProductAttributesLookup;
+
+use Automattic\WooCommerce\Utilities\ArrayUtil;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Data store class for the product attributes lookup table.
+ */
+class LookupDataStore {
+
+	/**
+	 * The lookup table name.
+	 *
+	 * @var string
+	 */
+	private $lookup_table_name;
+
+	/**
+	 * LookupDataStore constructor.
+	 */
+	public function __construct() {
+		global $wpdb;
+
+		$this->lookup_table_name = $wpdb->prefix . 'wc_product_attributes_lookup';
+	}
+
+	/**
+	 * Insert or update the lookup data for a given product or variation.
+	 * If a variable product is passed the information is updated for all of its variations.
+	 *
+	 * @param int|WC_Product $product Product object or id.
+	 * @throws \Exception A variation object is passed.
+	 */
+	public function update_data_for_product( $product ) {
+		// TODO: For now data is always deleted and fully regenerated, existing data should be updated instead.
+
+		if ( ! is_a( $product, \WC_Product::class ) ) {
+			$product = WC()->call_function( 'wc_get_product', $product );
+		}
+
+		if ( $this->is_variation( $product ) ) {
+			throw new \Exception( "LookupDataStore::update_data_for_product can't be called for variations." );
+		}
+
+		$this->delete_lookup_table_entries_for( $product->get_id() );
+
+		if ( $this->is_variable_product( $product ) ) {
+			$this->create_lookup_table_entries_for_variable_product( $product );
+		} else {
+			$this->create_lookup_table_entries_for_simple_product( $product );
+		}
+	}
+
+	/**
+	 * Delete all the lookup table entries for a given product
+	 * (entries are identified by the "parent_or_product_id" field)
+	 *
+	 * @param int $product_id Simple product id, or main/parent product id for variable products.
+	 */
+	private function delete_lookup_table_entries_for( int $product_id ) {
+		global $wpdb;
+
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
```

Comment:
Maybe we don't need the phpcs comment here as this is prepared?

Reply:
It's actually necessary, PHPCS throws the dreaded "Use placeholders and $wpdb->prepare()" because of the contcatenation of the table name.